esx_service/vmdk_ops.py

```diff
@@ -268,11 +282,21 @@ def getVMDKBacking(vmdk_path):
         # vsan object.
         cmd = "{0} {1}".format(OBJ_TOOL_CMD, uuid)
         rc, out = RunCommand(cmd)
-        fpath = "/vmfs/devices/vsan/{0}".format(uuid)
+        fpath = getVSANDevFSPath(uuid)
```

Comment:
I think the fix is doing some redundant steps and it complicates it.
We already have all data in getVMDKBacking()
I'd suggest just return (fpath, devfs_path) from getVMKDBacking() - i.e `return flatBacking, None`, `return fpath, fpath` and `return None, None)` in 3 `return` in the functions. And then in formatVmdk() simply do 
```python
 backing, devfs_path = getVMDKBacking(vmdk_path)
 ...
  # Cleanup temporary devFS node:
  if devfs_path:
       try: 
            os.remove(dev_fs_path)
            logging.debug(...)
       except: 
            logging.exception(....)
```